neqo-bin/src/client/http09.rs

```diff
@@ -86,10 +87,33 @@ impl super::Handler for Handler<'_> {
                     }
                 }
                 ConnectionEvent::StateChange(
-                    State::WaitInitial | State::Handshaking | State::Connected,
+                    State::WaitInitial | State::Handshaking | State::Connected | State::Confirmed,
                 ) => {
                     qdebug!("{event:?}");
                     self.download_urls(client);
```

Comment:
Why do we need to execute `self.download_urls` on `State::Confirmed`? In other words, why is the migration logic below not in its own `ConnectionEvent::StateChange(State::Confirmed)` match arm?

Reply:
I had a [similar comment](https://github.com/mozilla/neqo/pull/2180#discussion_r1802166587).

Reply:
That makes more sense, thanks.